### PR TITLE
Simplify code generation and merge integral and expression generators

### PR DESCRIPTION
**Pull request by IgorBaratta**
- Add shape to symbols
 - Store all symbols in `CodeRepresentation`
 - Use free functions for definitions and access
 -  Remove code for custom integrals (most of the code is untested and hasn't been updated for long time)

### CODE REVIEW

File: ffcx/ir/analysis/graph.py [hunk 1]
```diff
@@ -48,6 +48,14 @@ def add_edge(self, node1, node2):
         self.out_edges[node1] += [node2]
         self.in_edges[node2] += [node1]
 
+    def get_mode(self, mode: str):
+        """Return a dictionary of nodes with the given mode."""
```
Comment (chrisrichardson) [inline ffcx/ir/analysis/graph.py:52]: What are the possible modes?

File: ffcx/codegeneration/lnodes.py [hunk 2]
```diff
@@ -327,7 +331,7 @@ def __hash__(self):
         return hash(self.name)
 
     def __repr__(self):
-        return self.name
+        return self.name + (f"[{self.shape}]" if self.shape else "") + (f":{self.dtype}" if self.dtype else "")
```
Comment (chrisrichardson) [inline ffcx/codegeneration/lnodes.py:334]: dtype should always be set

File: ffcx/ir/analysis/graph.py (hunk 1 above)
Comment (IgorBaratta) [reply ffcx/ir/analysis/graph.py:52]: "piecewise" and "varying"

Comment (chrisrichardson) [reply ffcx/ir/analysis/graph.py:52]: Just add to docs...

**Comment by jhale** [comment]
This has moved quite far from `main`.

**Comment by IgorBaratta** [comment]
I'm closing this now. I've initiated work on a new branch, aiming for fewer modifications thanks to @jorgensd's PR.

**Pull request closed without merging**